<ui version="4.0" >
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>Circuit Editor</string>
  </property>
  <widget class="QWidget" name="centralwidget" >
   <property name="geometry" >
    <rect>
     <x>286</x>
     <y>48</y>
     <width>514</width>
     <height>383</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout" >
    <item>
     <widget class="CircuitWidget" native="1" name="circuitWidget" />
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar" >
   <property name="geometry" >
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile" >
    <property name="title" >
     <string>File</string>
    </property>
    <addaction name="actionOpen" />
    <addaction name="actionSave" />
    <addaction name="separator" />
    <addaction name="actionQuit" />
   </widget>
   <addaction name="menuFile" />
  </widget>
  <widget class="QStatusBar" name="statusbar" >
   <property name="geometry" >
    <rect>
     <x>0</x>
     <y>578</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="toolBar" >
   <property name="geometry" >
    <rect>
     <x>0</x>
     <y>22</y>
     <width>800</width>
     <height>26</height>
    </rect>
   </property>
   <property name="windowTitle" >
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea" >
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak" >
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen" />
   <addaction name="actionSave" />
   <addaction name="separator" />
   <addaction name="actionConnect" />
   <addaction name="actionDisconnect" />
   <addaction name="actionFlip" />
   <addaction name="actionDestroy" />
   <addaction name="actionCancel" />
  </widget>
  <widget class="QDockWidget" name="dockWidget" >
   <property name="geometry" >
    <rect>
     <x>0</x>
     <y>48</y>
     <width>280</width>
     <height>383</height>
    </rect>
   </property>
   <property name="floating" >
    <bool>true</bool>
   </property>
   <property name="windowTitle" >
    <string>Object Palette</string>
   </property>
   <attribute name="dockWidgetArea" >
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents" >
    <property name="geometry" >
     <rect>
      <x>0</x>
      <y>26</y>
      <width>280</width>
      <height>357</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2" >
     <item>
      <widget class="QTreeWidget" name="objectTree" >
       <property name="dragEnabled" >
        <bool>true</bool>
       </property>
       <property name="dragDropMode" >
        <enum>QAbstractItemView::DragOnly</enum>
       </property>
       <column>
        <property name="text" >
         <string>Circuit Objects</string>
        </property>
       </column>
       <item>
        <property name="text" >
         <string>Nodes</string>
        </property>
        <item>
         <property name="text" >
          <string>Normal node</string>
         </property>
        </item>
       </item>
       <item>
        <property name="text" >
         <string>Linear</string>
        </property>
        <item>
         <property name="text" >
          <string>Resistor</string>
         </property>
        </item>
        <item>
         <property name="text" >
          <string>Voltage source</string>
         </property>
        </item>
        <item>
         <property name="text" >
          <string>Current source</string>
         </property>
        </item>
       </item>
       <item>
        <property name="text" >
         <string>Diodes</string>
        </property>
        <item>
         <property name="text" >
          <string>Shockley diode</string>
         </property>
        </item>
       </item>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_2" >
   <property name="geometry" >
    <rect>
     <x>0</x>
     <y>437</y>
     <width>800</width>
     <height>141</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Minimum" hsizetype="Minimum" >
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize" >
    <size>
     <width>300</width>
     <height>122</height>
    </size>
   </property>
   <property name="floating" >
    <bool>false</bool>
   </property>
   <property name="windowTitle" >
    <string>Analysis</string>
   </property>
   <attribute name="dockWidgetArea" >
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2" >
    <property name="geometry" >
     <rect>
      <x>0</x>
      <y>26</y>
      <width>800</width>
      <height>115</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_4" >
     <item>
      <spacer name="horizontalSpacer_2" >
       <property name="orientation" >
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0" >
        <size>
         <width>209</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout" >
       <item row="0" column="0" >
        <widget class="QLabel" name="label" >
         <property name="text" >
          <string>Analysis type:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1" >
        <widget class="QComboBox" name="analysisCombo" >
         <item>
          <property name="text" >
           <string>Operating point</string>
          </property>
         </item>
         <item>
          <property name="text" >
           <string>Thevenin equivalent</string>
          </property>
         </item>
         <item>
          <property name="text" >
           <string>Norton equivalent</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="1" column="0" >
        <widget class="QLabel" name="label_2" >
         <property name="text" >
          <string>Outputs:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1" >
        <layout class="QHBoxLayout" name="horizontalLayout_3" >
         <item>
          <widget class="QCheckBox" name="doSummaryCheck" >
           <property name="text" >
            <string>Summary table</string>
           </property>
           <property name="checked" >
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="setAttributesCheck" >
           <property name="text" >
            <string>Set attributes</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="2" column="1" >
        <widget class="QPushButton" name="beginButton" >
         <property name="sizePolicy" >
          <sizepolicy vsizetype="Maximum" hsizetype="Minimum" >
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text" >
          <string>Begin</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <spacer name="horizontalSpacer" >
       <property name="orientation" >
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0" >
        <size>
         <width>208</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionOpen" >
   <property name="text" >
    <string>Open</string>
   </property>
   <property name="shortcut" >
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave" >
   <property name="text" >
    <string>Save</string>
   </property>
   <property name="shortcut" >
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionQuit" >
   <property name="text" >
    <string>Quit</string>
   </property>
  </action>
  <action name="actionConnect" >
   <property name="text" >
    <string>Connect</string>
   </property>
   <property name="shortcut" >
    <string>C</string>
   </property>
   <property name="shortcutContext" >
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionDisconnect" >
   <property name="text" >
    <string>Disconnect</string>
   </property>
   <property name="shortcut" >
    <string>D</string>
   </property>
   <property name="shortcutContext" >
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionFlip" >
   <property name="text" >
    <string>Flip</string>
   </property>
   <property name="shortcut" >
    <string>F</string>
   </property>
   <property name="shortcutContext" >
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionDestroy" >
   <property name="text" >
    <string>Destroy</string>
   </property>
   <property name="shortcut" >
    <string>Backspace</string>
   </property>
   <property name="shortcutContext" >
    <enum>Qt::WindowShortcut</enum>
   </property>
  </action>
  <action name="actionCancel" >
   <property name="enabled" >
    <bool>false</bool>
   </property>
   <property name="text" >
    <string>Cancel</string>
   </property>
   <property name="shortcut" >
    <string>Esc</string>
   </property>
   <property name="shortcutContext" >
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CircuitWidget</class>
   <extends>QWidget</extends>
   <header>circuit_gui.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
